storage/range.go

```diff
@@ -1246,6 +1251,22 @@ func (r *Range) InternalMerge(batch engine.Engine, ms *engine.MVCCStats, args *p
 	reply.SetGoError(err)
 }
 
+// InternalTruncateLog discards a prefix of the raft log.
+func (r *Range) InternalTruncateLog(batch engine.Engine, ms *engine.MVCCStats, args *proto.InternalTruncateLogRequest, reply *proto.InternalTruncateLogResponse) {
+	kvs, err := engine.MVCCScan(batch,
```

Comment:
ClearRange doesn't work with an engine.Batch. As we discussed offline I've just copied the code from ClearRange.